src/salary.cairo

```diff
@@ -0,0 +1,336 @@
+// @title Mesh Salary contract Cairo 2.2
+// @author Mesh Finance
+// @license MIT
+// @notice Contract to disburse salary to contibutors
+
+use starknet::ContractAddress;
+use array::Array;
+
+#[derive(Drop, Serde, starknet::Store)]
+struct Salary {
+    // @notice contributor salary earned so far
+    cum_salary: u256,
+    // @notice salary claimed so far
+    claimed_salary: u256
+}
+
+//
+// External Interfaces
+//
+
+#[starknet::interface]
+trait IERC20<T> {
+    fn balance_of(self: @T, account: ContractAddress) -> u256;
+    fn balanceOf(self: @T, account: ContractAddress) -> u256; // TODO Remove after regenesis
+    fn transfer(ref self: T, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(ref self: T, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn transferFrom(ref self: T, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool; // TODO Remove after regenesis
+}
+
+#[starknet::interface]
+trait IGuild<T> {
+    fn get_contribution_tier(self: @T, contributor: ContractAddress) -> u32;
+}
+
+#[starknet::interface]
+trait IMaster<T> {
+    // fn get_total_contribution(self: @T, month_id: u32) -> TotalMonthlyContribution;
+    // fn get_contributions_points(self: @T, contributor: ContractAddress) -> Contribution;
+    fn get_contributions_data(self: @T, contributor: ContractAddress, guild: felt252) -> Array<u32>;
+    fn get_guild_total_contribution(self: @T, month_id: u32, guild: felt252) -> u32;
+    fn get_guild_points(self: @T, contributor: ContractAddress, guild: felt252) -> u32;
+
+
+}
+
+//
+// Contract Interface
+//
+#[starknet::interface]
+trait ISalaryDistributor<TContractState> {
+    // view functions
+    fn token(self: @TContractState) -> ContractAddress;
+    fn master(self: @TContractState) -> ContractAddress;
+    fn get_cum_salary(self: @TContractState, contributor: ContractAddress) -> u256;
```

Comment:
`get_cum_salary_by_month` -> maybe, if frontend need it can add it.
`set_master` -> added
`set_token` -> can't change token as user claiming prior salary, will withdraw in new tokens as well. Although we can have multiple tokens and set which month salary will be payout in which tokens, but it will just complicate without adding any value.